Src/FluentAssertions/Specialized/AsyncFunctionAssertionsBase.cs

```diff
@@ -11,67 +11,20 @@ namespace FluentAssertions.Specialized;
 /// <summary>
 /// Contains a number of methods to assert that an asynchronous method yields the expected result.
 /// </summary>
+/// <typeparam name="TTask">The type of <see cref="Task{T}"/> to be handled.</typeparam>
+/// <typeparam name="TAssertions">The type of assertion to be returned.</typeparam>
 [DebuggerNonUserCode]
-public class AsyncFunctionAssertions<TTask, TAssertions> : DelegateAssertionsBase<Func<TTask>, TAssertions>
+public class AsyncFunctionAssertionsBase<TTask, TAssertions> : DelegateAssertionsBase<Func<TTask>, TAssertions>
```

Comment:
For visibility in the PR, here's the current inheritance hierarchy.
```
DelegateAssertionsBase<TDelegate, TAssertions>
	DelegateAssertions<TDelegate, TAssertions>
		ActionAssertions
		FunctionAssertions<T>
	AsyncFunctionAssertions<TTask, TAssertions>
		GenericAsyncFunctionAssertions<TResult>
		NonGenericAsyncFunctionAssertions
```

Should we just keep the current name `AsyncFunctionAssertions`?
I assume the `Base` suffix for `DelegateAssertionsBase` is just a necessity to avoid naming conflict with its derived `DelegateAssertions`.

Reply:
You are right, that way it is not that consistent.
I'll revert this change here and rethink the class hierarchy separately.